pkg/3scale/amp/operator/redis_options_provider.go

```diff
@@ -270,3 +271,12 @@ func (r *RedisOptionsProvider) setTopologySpreadConstraints() {
 		r.options.BackendRedisTopologySpreadConstraints = r.apimanager.Spec.Backend.RedisTopologySpreadConstraints
 	}
 }
+
+func (r *RedisOptionsProvider) setPodTemplateAnnotations() {
+	if r.apimanager.Spec.System != nil {
```

Comment:
Can `r.apimanager.Spec.System` be nil? I expected to be set in defaults when not in the CR.

Reply:
yes, I see ... setSystemSpecDefaults(..).   Thank you very much. I will check/update

Reply:
fixed, also in searchd. Thank you